### PR TITLE
[threads] Return boolean instead of pointer from Thread_internal

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -40,7 +40,7 @@ MONO_VERSION_BUILD=`echo $VERSION | cut -d . -f 3`
 # This can be reset to 0 when Mono's version number is bumped
 # since it's part of the corlib version (the prefix '1' in the full
 # version number is to ensure the number isn't treated as octal in C)
-MONO_CORLIB_COUNTER=3
+MONO_CORLIB_COUNTER=1242018
 MONO_CORLIB_VERSION=`printf "1%02d%02d%02d%03d" $MONO_VERSION_MAJOR $MONO_VERSION_MINOR 0 $MONO_CORLIB_COUNTER`
```
Comment (vargaz) [inline configure.ac]: I think this needs to stay a small number.

Comment (jaykrell) [reply configure.ac]: Yeah I see the %03d, I changed it to 4.

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 mcs

**Comment by jaykrell** [comment]
@monojenkins Linux AArch64 Interpreter

**Comment by jaykrell** [comment]
@monojenkins Linux ARMv7 Interpreter

**Comment by luhenry** [comment]
@monojenkins build pkg with monolite

**Comment by akoeplinger** [comment]
@monojenkins build deb with monolite

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 mcs

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 Interpreter

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 Interpreter

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64

**Comment by luhenry** [comment]
@monojenkins build Windows i386

**Comment by luhenry** [comment]
@monojenkins build Windows x64

**Pull request merged**